A recent change to CI (as this wasn't causing issues in prior PR's) is causing a build failure because the Digital Twins Model Parser does not have a strong name (its not signed).

Error from CI pipeline
```
 CSC : error CS8002: Referenced assembly 'Microsoft.Azure.DigitalTwins.Parser, Version=3.12.5.0, Culture=neutral, PublicKeyToken=null' does not have a strong name.
```

--- a/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ModelResolutionSamples.cs
+++ b/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ModelResolutionSamples.cs
@@ -48,14 +48,13 @@ namespace Azure.Iot.ModelsRepository.Samples
         {
             #region Snippet:ModelsRepositorySamplesGetModelsFromGlobalRepoAsync
 
-            var dtmi = "dtmi:com:example:TemperatureController;1";
-
             // Global endpoint client
             var client = new ModelsRepositoryClient();
 
             // The output of GetModelsAsync() will include at least the definition for the target dtmi.
             // If the dependency model resolution option is not disabled, then models in which the
             // target dtmi depends on will also be included in the returned IDictionary<string, string>.
+            var dtmi = "dtmi:com:example:TemperatureController;1";
             IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 
             // In this case the above dtmi has 2 model dependencies.
@@ -91,14 +90,13 @@ namespace Azure.Iot.ModelsRepository.Samples
         {
             #region Snippet:ModelsRepositorySamplesGetModelsFromLocalRepoAsync
 
-            var dtmi = "dtmi:com:example:TemperatureController;1";
-
             // Local sample repository client
             var client = new ModelsRepositoryClient(new Uri(ClientSamplesLocalModelsRepository));
 
             // The output of GetModelsAsync() will include at least the definition for the target dtmi.
             // If the dependency model resolution option is not disabled, then models in which the
             // target dtmi depends on will also be included in the returned IDictionary<string, string>.
+            var dtmi = "dtmi:com:example:TemperatureController;1";
             IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 
             // In this case the above dtmi has 2 model dependencies.
@@ -110,8 +108,8 @@ namespace Azure.Iot.ModelsRepository.Samples
 
         public static async Task TryGetModelsFromGlobalRepoButNotFoundAsync()
         {
-            var dtmi = "dtmi:com:example:NotFound;1";
             var client = new ModelsRepositoryClient();
+            var dtmi = "dtmi:com:example:NotFound;1";
 
             try
             {
@@ -126,8 +124,8 @@ namespace Azure.Iot.ModelsRepository.Samples
 
         public static async Task TryGetModelsFromLocalRepoButNotFoundAsync()
         {
-            var dtmi = "dtmi:com:example:NotFound;1";
             var client = new ModelsRepositoryClient(new Uri(ClientSamplesLocalModelsRepository));
+            var dtmi = "dtmi:com:example:NotFound;1";
 
             try
             {

--- a/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ParserIntegrationSamples.cs
+++ b/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ParserIntegrationSamples.cs
@@ -15,8 +15,8 @@ namespace Azure.Iot.ModelsRepository.Samples
         {
             #region Snippet:ModelsRepositorySamplesParserIntegrationGetModelsAndParseAsync
 
-            var dtmi = "dtmi:com:example:TemperatureController;1";
             var client = new ModelsRepositoryClient();
+            var dtmi = "dtmi:com:example:TemperatureController;1";
             IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
             var parser = new ModelParser();
             IReadOnlyDictionary<Dtmi, DTEntityInfo> parseResult = await parser.ParseAsync(models.Values.ToArray());
@@ -29,8 +29,8 @@ namespace Azure.Iot.ModelsRepository.Samples
         {
             #region Snippet:ModelsRepositorySamplesParserIntegrationParseAndGetModelsAsync
 
-            var dtmi = "dtmi:com:example:TemperatureController;1";
             var client = new ModelsRepositoryClient(new ModelsRepositoryClientOptions(resolutionOption: DependencyResolutionOption.Disabled));
+            var dtmi = "dtmi:com:example:TemperatureController;1";
             IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
             var parser = new ModelParser
             {

--- a/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/readme.md
+++ b/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/readme.md
@@ -1,10 +1,10 @@
-# Introduction
+# IoT Models Repository Samples
 
 The Azure IoT Models Repository enables builders to manage and share digital twin models for global consumption. The models are [JSON-LD][json_ld_reference] documents defined using the Digital Twins Definition Language ([DTDL][dtdlv2_reference]).
 
 For more info about the Azure IoT Models Repository checkout the [docs][modelsrepository_msdocs].
 
-# IoT Models Repository Samples
+## Introduction
 
 You can explore the models repository APIs with the client library using the samples project.
 
@@ -15,8 +15,6 @@ The samples project demonstrates the following:
 - Integration with the Digital Twins Model Parser
 
 ## Initializing the models repository client
-
-### Initialization
 
 ```C# Snippet:ModelsRepositorySamplesCreateServiceClientWithGlobalEndpoint
 // When no URI is provided for instantiation, the Azure IoT Models Repository global endpoint

--- a/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/readme.md
+++ b/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/readme.md
@@ -49,14 +49,13 @@ Publishing models to the models repository requires [exercising][modelsrepositor
 After publishing, your model(s) will be available for consumption from the global repository endpoint. The following snippet shows how to retrieve the corresponding JSON-LD content.
 
 ```C# Snippet:ModelsRepositorySamplesGetModelsFromGlobalRepoAsync
-var dtmi = "dtmi:com:example:TemperatureController;1";
-
 // Global endpoint client
 var client = new ModelsRepositoryClient();
 
 // The output of GetModelsAsync() will include at least the definition for the target dtmi.
 // If the dependency model resolution option is not disabled, then models in which the
 // target dtmi depends on will also be included in the returned IDictionary<string, string>.
+var dtmi = "dtmi:com:example:TemperatureController;1";
 IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 
 // In this case the above dtmi has 2 model dependencies.
@@ -64,17 +63,18 @@ IDictionary<string, string> models = await client.GetModelsAsync(dtmi).Configure
 Console.WriteLine($"{dtmi} resolved in {models.Count} interfaces.");
 ```
 
-Because GitHub pull-request workflows are a core aspect of the IoT Models Repository service, and to supported related use cases, the client supports getting models from a local repository.
+GitHub pull-request workflows are a core aspect of the IoT Models Repository service. To submit models, the user is expected to fork and clone the global [models repository project][modelsrepository_github_repo] then iterate against the local copy. Changes would then be pushed to the fork (ideally in a new branch) and a PR created against the global repository.
+
+To support this workflow and similar use cases, the client supports initialization with a local file-system URI. You can use this for example, to test and ensure newly added models to the locally cloned models repository are in their proper locations.
 
 ```C# Snippet:ModelsRepositorySamplesGetModelsFromLocalRepoAsync
-var dtmi = "dtmi:com:example:TemperatureController;1";
-
 // Local sample repository client
 var client = new ModelsRepositoryClient(new Uri(ClientSamplesLocalModelsRepository));
 
 // The output of GetModelsAsync() will include at least the definition for the target dtmi.
 // If the dependency model resolution option is not disabled, then models in which the
 // target dtmi depends on will also be included in the returned IDictionary<string, string>.
+var dtmi = "dtmi:com:example:TemperatureController;1";
 IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 
 // In this case the above dtmi has 2 model dependencies.
@@ -109,8 +109,8 @@ The samples provide two different patterns to integrate with the Digital Twins M
 The following snippet shows first fetching model definitions from the Azure IoT Models Repository then parsing them.
 
 ```C# Snippet:ModelsRepositorySamplesParserIntegrationGetModelsAndParseAsync
-var dtmi = "dtmi:com:example:TemperatureController;1";
 var client = new ModelsRepositoryClient();
+var dtmi = "dtmi:com:example:TemperatureController;1";
 IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 var parser = new ModelParser();
 IReadOnlyDictionary<Dtmi, DTEntityInfo> parseResult = await parser.ParseAsync(models.Values.ToArray());
@@ -121,8 +121,8 @@ Alternatively, the following snippet shows parsing a model, then fetching depend
 This is achieved by configuring the `ModelParser` to use the sample [ParserDtmiResolver][modelsrepository_sample_extension] client extension.
 
 ```C# Snippet:ModelsRepositorySamplesParserIntegrationParseAndGetModelsAsync
-var dtmi = "dtmi:com:example:TemperatureController;1";
 var client = new ModelsRepositoryClient(new ModelsRepositoryClientOptions(resolutionOption: DependencyResolutionOption.Disabled));
+var dtmi = "dtmi:com:example:TemperatureController;1";
 IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
 var parser = new ModelParser
 {
@@ -134,6 +134,7 @@ Console.WriteLine($"{dtmi} resolved in {models.Count} interfaces with {parseResu
 ```
 
 <!-- LINKS -->
+[modelsrepository_github_repo]: https://github.com/Azure/iot-plugandplay-models
 [modelsrepository_sample_extension]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ModelsRepositoryClientExtensions.cs
 [modelsrepository_clientoptions]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelsRepositoryClientOptions.cs
 [modelsrepository_msdocs]: https://docs.microsoft.com/azure/iot-pnp/concepts-model-repository

